### PR TITLE
[DOCS beta] clarify documentation for makeArray

### LINKED ISSUE #14741: `makeArray` does not work as commented
In `make-array`, it is commented as followed:

```javascript
 let controller = Ember.ArrayProxy.create({ content: [] });

 Ember.makeArray(controller) === controller;  // true
```

However it is not true. It seems `Ember.makeArray(controller)` is using `Array.isArray` now, which does not handle array-like object.

### PR DESCRIPTION
**Pull request by ghost**
Resolve #14741

### CODE REVIEW

File: packages/ember-utils/lib/make-array.js [hunk 1]
```diff
@@ -5,7 +5,8 @@ const { isArray } = Array;
 /**
  Forces the passed object to be part of an array. If the object is already
  an array, it will return the object. Otherwise, it will add the object to
- an array. If obj is `null` or `undefined`, it will return an empty array.
+ an array. If object is `null` or `undefined`, it will return an empty array.
+ `makeArray` treats `ArrayProxy` like any other object.
```
[the remark this answers is not in the document]
Comment (efx) [reply packages/ember-utils/lib/make-array.js]: Certainly.

File: packages/ember-utils/lib/make-array.js [hunk 2]
```diff
@@ -19,7 +20,7 @@ const { isArray } = Array;
 
  let controller = ArrayProxy.create({ content: [] });
```
[the remark this answers is not in the document]
Comment (efx) [reply packages/ember-utils/lib/make-array.js]: ah, good point. will do.

**Review by rwjblue — CHANGES_REQUESTED** [review]
Sorry for the run around here, I realize that my previous comment was very vague around _which_ thing to rename...

**Comment by efx** [comment]
No problemo! In documentation right wording is essential.

**Review by rwjblue — APPROVED** [review]

**Pull request merged**